### PR TITLE
replace lazy_static with once_cell

### LINKED ISSUE #576: Replace lazy_static with once_cell
We're currently using `lazy_static` in some test code. Since it's a macro, it looks and feels a little magic. There has been a less magic alternative for a while in the form of the `once_cell` crate (whose interface is also available in `std` in nightly). We should switch to it for better readability.

### PR DESCRIPTION
**Pull request by cutecutecat**
fix #576 

Signed-off-by: cutecutecat <starkind1997@gmail.com>

### CODE REVIEW

File: crates/matrix-sdk-sled/src/cryptostore.rs [hunk 1]
```diff
@@ -1100,15 +1100,14 @@ impl CryptoStore for SledStore {
...
+    use once_cell::sync::Lazy;
+
     use tempfile::{tempdir, TempDir};
 
     use super::SledStore;
-    lazy_static! {
-        /// This is an example for using doc comment attributes
-        static ref TMP_DIR: TempDir = tempdir().unwrap();
-    }
+    /// This is an example for using doc comment attributes
```
Comment (jplatte) [inline crates/matrix-sdk-sled/src/cryptostore.rs:1109]: You can remove this, it seems like somebody previous copy-pasted from an example and forgot to remove it.

File: crates/matrix-sdk-test/src/test_json/events.rs [hunk 2]
```diff
@@ -1,24 +1,22 @@
...
-        "sender": "@example:localhost",
-        "state_key": "",
-        "type": "m.room.canonical_alias",
-        "unsigned": {
-            "age": 703422
-        }
-    });
-}
+pub static ALIAS: Lazy<JsonValue> = Lazy::new(|| {
+    json!({"content": {
```
Comment (jplatte) [inline crates/matrix-sdk-test/src/test_json/events.rs:5]: Please put the `"content": {` part on its own line like before and clean up the indentation.

File: crates/matrix-sdk-test/Cargo.toml [hunk 3]
```diff
@@ -16,7 +16,7 @@ appservice = []
 
 [dependencies]
 http = "0.2.4"
-lazy_static = "1.4.0"
+once_cell = "1.10.0"
```
[the remark this answers is not in the document]
Comment (jplatte) [reply crates/matrix-sdk-test/Cargo.toml:19]: Same in the other files, would be good to have alphabtical ordering.

**Review by jplatte — COMMENTED** [review]
Thanks for the PR! Looks good overall, there's just a few small things I'd like to see fixed before merging.

**Comment by gnunicorn** [comment]
@cutecutecat mind the style complaints?

**Review by gnunicorn — APPROVED** [review]

File: crates/matrix-sdk-test/src/test_json/events.rs [hunk 4]
```diff
@@ -31,12 +30,11 @@ lazy_static! {
         "type": "m.room.aliases",
         "unsigned": {
             "age": 703422
-        }
-    });
-}
+    }})
```
Comment (jplatte) [inline crates/matrix-sdk-test/src/test_json/events.rs:33]: Same here:

```suggestion
        }
    })
```

File: crates/matrix-sdk-test/src/test_json/events.rs [hunk 5]
```diff
@@ -1,24 +1,23 @@
...
+        "alias": "#tutorial:localhost"
+    },
+    "event_id": "$15139375513VdeRF:localhost",
+    "origin_server_ts": 151393755,
+    "sender": "@example:localhost",
+    "state_key": "",
+    "type": "m.room.canonical_alias",
+    "unsigned": {
+        "age": 703422
+    }})
```
Comment (jplatte) [inline crates/matrix-sdk-test/src/test_json/events.rs:16]:
```suggestion
            "alias": "#tutorial:localhost"
        },
        "event_id": "$15139375513VdeRF:localhost",
        "origin_server_ts": 151393755,
        "sender": "@example:localhost",
        "state_key": "",
        "type": "m.room.canonical_alias",
        "unsigned": {
            "age": 703422
        }
    })
```

**Comment by jplatte** [comment]
Oh, there's also merge conflicts now apparently. Sorry about that! I can finish the small remaining bits if you want.

File: crates/matrix-sdk-test/src/test_json/events.rs [hunk 6]
```diff
@@ -1,9 +1,9 @@
-use lazy_static::lazy_static;
+use once_cell::sync::Lazy;
 use serde_json::{json, Value as JsonValue};
 
-lazy_static! {
-    pub static ref ALIAS: JsonValue = json!({
-        "content": {
+pub static ALIAS: Lazy<JsonValue> = Lazy::new(|| {
+    json!({
+            "content": {
```
Comment (jplatte) [inline crates/matrix-sdk-test/src/test_json/events.rs:6]:
```suggestion
        "content": {
```

File: crates/matrix-sdk-test/src/test_json/events.rs [hunk 7]
```diff
@@ -14,29 +14,29 @@ lazy_static! {
...
+                    "#tutorial:localhost"
+                ]
+            },
+            "event_id": "$15139375516NUgtD:localhost",
+            "origin_server_ts": 151393755,
+            "sender": "@example:localhost",
+            "state_key": "localhost",
+            "type": "m.room.aliases",
+            "unsigned": {
+                "age": 703422
```
Comment (jplatte) [inline crates/matrix-sdk-test/src/test_json/events.rs:33]:
```suggestion
        "content": {
            "aliases": [
                "#tutorial:localhost"
            ]
        },
        "event_id": "$15139375516NUgtD:localhost",
        "origin_server_ts": 151393755,
        "sender": "@example:localhost",
        "state_key": "localhost",
        "type": "m.room.aliases",
        "unsigned": {
            "age": 703422
```

**Pull request merged**